src/core/Akka.Streams/Stage/GraphStage.cs

```diff
@@ -1746,6 +1827,35 @@ protected internal virtual void AfterPostStop()
                 _stageActor.Stop();
                 _stageActor = null;
             }
+
+            var inProgress = AsyncCallbacksInProgress.GetAndSet(null);
+            if (inProgress is not null && inProgress.Count > 0)
+            {
+                try
+                {
+                    throw new StreamDetachedException($"Stage with GraphStageLogic [{this}] stopped before async invocation was processed");
+                }
+                catch (Exception ex)
+                {
+                    foreach (var tcs in inProgress)
+                    {
+                        tcs.TrySetException(ex);
+                    }
+                }
+            }
+        }
+
+        internal void OnFeedbackDispatched(TaskCompletionSource<Done> p)
+        {
+            switch (AsyncCallbacksInProgress.Value)
+            {
+                case null:
+                    // already finished, nothing to do here
+                    break;
+                case var x:
+                    x.Remove(p);
+                    break;
+            } 
```

Comment:
When interpreter finished processing the callback, it signals the logic (this instance) through this callback. The logic then removes the `TaskCompletionSource` from the pending list.